### PR TITLE
[RDBMS] Python SDK migration

### PR DESCRIPTION
**Pull request by DaeunYim**
**Description**

Upgrading the Python SDK version for rdbms to GA version (8.0.0)

**Testing Guide**

azdev test rdbms 

**History Notes**

Upgraded Python SDK version to 8.0.0

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

File: scripts/sdk_process/patch_models.py
Comment (jiasli) [inline scripts/sdk_process/patch_models.py:235]: To CLI team: Why does `azure-mgmt-rdbms` contain dashes (`-`) instead of dots (`.`)? We need a unified and verified convention.

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/recordings/test_mysql_flexible_server_vnet_server_create.yaml
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/recordings/test_mysql_flexible_server_vnet_server_create.yaml:21]: Please merge from `dev` and re-run this test after #17185 gets merged.

File: scripts/sdk_process/patch_models.py
Comment (DaeunYim) [reply scripts/sdk_process/patch_models.py]: Thanks for pointing this out!

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/recordings/test_mysql_flexible_server_vnet_server_create.yaml
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/recordings/test_mysql_flexible_server_vnet_server_create.yaml:21]: Got it!

**Comment by DaeunYim** [comment]
Hi :) Is there any other comments for this PR?

File: scripts/sdk_process/patch_models.py
Comment (evelyn-ys) [inline scripts/sdk_process/patch_models.py:235]:
```suggestion
        'azure.mgmt.rdbms'
```
@jsntcy to check

File: src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py:41]: There're huge changes from track1 credentials to track2 credential. I'm afraid the origin credential constructor can't work with track2. Have you tested?

File: src/azure-cli/azure/cli/command_modules/rdbms/_flexible_server_util.py
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/_flexible_server_util.py:112]: Should we return `firewall.result().name` to keep align with before?

File: src/azure-cli/azure/cli/command_modules/rdbms/_util.py
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/_util.py:98]:
```suggestion
    if db_engine == 'mysql':
        firewall_client = cf_mysql_firewall_rules(cmd.cli_ctx, None)
    elif db_engine == 'mariadb':
        firewall_client = cf_mariadb_firewall_rules(cmd.cli_ctx, None)
    parameters = {'name': firewall_name, 'start_ip_address': start_ip, 'end_ip_address': end_ip}
```

Since all `FirewallRule`s inherit `msrest.serialization.Model`, you can define plain `dict` and it will be serialized automatically. This will make your code cleaner.

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:118]: Why remove `id_part='child_name_1'` here? This will disable customers using resource id...

File: src/azure-cli/azure/cli/command_modules/rdbms/custom.py
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/custom.py:840]: `create_or_update` and `begin_create_or_update` both return `LROPoller`, why drop `.result()` in latest code?

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py:94]: When `begin_delete` throw exception, is `result.result()` still needed? 

Is it better to have `result = client.begin_delete(...).result()` inside `try...except`?

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py:106]: Why this func doesn't need `.result()`?

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py:135]: Same as https://github.com/Azure/azure-cli/pull/17191/files#r589954953

File: src/azure-cli/azure/cli/command_modules/rdbms/_flexible_server_util.py
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_flexible_server_util.py:112]: good catch,,, thanks..!

File: src/azure-cli/azure/cli/command_modules/rdbms/_util.py
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_util.py:98]: Thanks

File: src/azure-cli/azure/cli/command_modules/rdbms/commands.py
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/commands.py:392]: oh this is typo..

Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/commands.py:392]: this is a typo,, thanks..

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py:94]: Exactly. When `LROPoller` is returned to Azure CLI core, there is no need to call `.result()` as the core will handle it for you.

Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py:94]: Made the change. Thanks a lot!!

File: src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py:41]: Sorry, I totally missed this point. 

Will this migration include azure-identity library? 

Looks like now the python sdk suggests using ClientSecretCredential instead of ServicePrincipalCredentials (https://docs.microsoft.com/en-us/azure/developer/python/azure-sdk-authenticate?tabs=cmd) . Testing with ServicePrincipalCredentials didn't work but ClientSecretCredential worked well.. 

Let me know if I am allowed to replace ServicePrincipalCredentials with ClientSecretCredential..

Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py:41]: @jiasli is expert of cli auth. Maybe he can answer this question😃

File: scripts/sdk_process/patch_models.py
Comment (evelyn-ys) [reply scripts/sdk_process/patch_models.py:235]: Why use `-` between `azure` and `mgmt`, but `.` between `mgmt` and `rdbms`? I'm little confused

File: src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py:41]: Yes, when running with the ServicePrincipalCredentials, the error I was seeing was 
AttributeError: 'ServicePrincipalCredentials' object has no attribute 'get_token'..!

**Review by jiasli — APPROVED** [review]
Approve for core.

File: src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py:41]: Thank you so much for detailed explanation! Made the change per your comment

Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py:41]: Sorry to bother @jiasli but the style error is telling me 
src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py:29:12: E0611: No name 'identity' in module 'azure' (no-name-in-module)
https://dev.azure.com/azure-sdk/public/_build/results?buildId=776163&view=logs&j=36dd4138-4d53-5e46-00d9-e5cd9744be05&t=1cf3879c-0a42-5ccd-7693-7a4781d739d8&l=134

This import works and style test passes on my local,, so wanted to ask you if there's a known issue/ way to unblock this,,,!!!

Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py:41]: This is because `dev` branch of Azure CLI doesn't depend on Azure Identity, so it is not installed. Do you have to use Azure Identity for this feature? If so, I can add this dependency.

Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py:41]: @jiasli Thank you so much!

Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py:41]: RDBMS module uses get_mgmt_service_client but I think the predecessor also implemented a way to override the authentification using credential. I wanted to keep both ways in case users were using this approach.. thanks!!

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:137]: Wondering if `id_part='child_name_1'` is necessary. I noticed that `database_name` for other commands has `id_part=None`. What's the difference between this command group and others?

**Review by evelyn-ys — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_params.py:137]: Thanks for pointing this out.. I realized for single server the command is 'az mysql db' and for flexible server the command is 'az mysql flexible-server db' so single server database name shouldn't have child_name_1 as ID part. so removed this

For flexible server, database name needs to be child name but I think we set the id part as none for every name parameters to avoid linter issue..!

**Comment by DaeunYim** [comment]
CI tests passed with updated network module recording files (related to rdbms single servers)

**Review by kairu-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py:61]: Unless `AZURE_AUTHORITY_HOST` is specified, this doesn't work for sovereign clouds or Azure stacks, since authority of `cli_ctx` is not passed to `ClientSecretCredential`.

**Pull request merged**
